src/org/centos/contra/Infra/Utils.groovy

```diff
@@ -299,6 +300,83 @@ def createOpenstackInstances(HashMap<String, String>openstackData){
     return openstackInstances
 }
 
+/**
+ * Method to create a collection of instances of the OpenShift Cloud Platform provider type based on supplied data
+ * @param openshiftData
+ * @return
+ */
+def createOpenshiftInstances(HashMap<String, String>openshiftData){
+    ArrayList<Openshift> openshiftInstances = []
+
+    openshiftData.instances.each {LinkedHashMap<String, String> instance ->
+        // Setting default params from provider class constructor
+        instance.name = instance.name ?: openshiftData.name
+        instance.namespace = instance.namespace ?: openshiftData.namespace
+        instance.password = instance.password ?: openshiftData.password
+        instance.username = instance.username ?: openshiftData.username
+
+        // 1. set api* values here
+        // 2. check if value for Loadbalancer has to be set(neeed in provider class too)
+        // 3. check if keyPair is required for ocp
+        // 4. "" userData, user
+        def user = null
```

Comment:
Lines 322-325 can be rewritten as:

```` groovy
instance.user = instance.user ?: openshiftData.user
````

Reply:
good to know, ack, groovy truth, thanks!